core/include/scipp/core/element/event_operations.h

```diff
@@ -71,10 +74,16 @@ using args = std::tuple<Data, Coord, span<const Edge>, span<const Weight>>;
 } // namespace map_and_mul_detail
 
 constexpr auto map_and_mul = overloaded{
-    element::arg_list<map_and_mul_detail::args<double, double, double, double>,
-                      map_and_mul_detail::args<float, double, double, double>,
-                      map_and_mul_detail::args<float, double, double, float>,
-                      map_and_mul_detail::args<double, float, float, double>>,
+    element::arg_list<
+        map_and_mul_detail::args<double, double, double, double>,
+        map_and_mul_detail::args<double, double, double, float>,
+        map_and_mul_detail::args<float, double, double, double>,
+        map_and_mul_detail::args<float, double, double, float>,
+        map_and_mul_detail::args<double, float, float, double>,
+        map_and_mul_detail::args<double, time_point, time_point, double>,
+        map_and_mul_detail::args<double, time_point, time_point, float>,
+        map_and_mul_detail::args<float, time_point, time_point, double>,
+        map_and_mul_detail::args<float, time_point, time_point, float>>,
```

Comment:
No additional tests needed for these combinations?

Reply:
See second paragraph in PR description. We generally do not test for all possibly types a template accepts. If you talk about ensuring that the operations ensure these types, I think we need to find a more holistic solution, rather than adding specific tests in a couple of dozen places (for various functions).